flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/optimization/PredictionFunction.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.optimization
+
+import org.apache.flink.ml.common.WeightVector
+import org.apache.flink.ml.math.{Vector => FlinkVector, BLAS}
+
+/** An abstract class for prediction functions to be used in optimization **/
+abstract class PredictionFunction extends Serializable {
+  def predict(features: FlinkVector, weights: WeightVector): Double
+
+  def gradient(features: FlinkVector, weights: WeightVector): FlinkVector
+}
+
+/** A linear prediction function **/
+class LinearPrediction extends PredictionFunction {
+  override def predict(features: FlinkVector, weights: WeightVector): Double = {
+    BLAS.dot(features, weights.weights) + weights.intercept
```

Comment:
Shall we call the weights parameter differently? Now we have `weights.weights` which looks not so nice. Maybe `weightVector.weights`?